Execute a pipeline

mikerowehl opened the pull request:
Has an option to poll for completion.

imosquera committed:
Add a command to execute a pipeline and poll for completion

mikerowehl committed:
Remove some unused imports and types.

mikerowehl committed:
Fix up formatting

robzienert commented on actions.go:
```diff
@@ -15,6 +15,52 @@ import (
 )
 
 // PipelineSaveAction creates the ActionFunc for saving pipeline configurations.
+func PipelineExecAction(clientConfig spinnaker.ClientConfig) cli.ActionFunc {
```
I merged a doc PR just before yours, so this function needs a comment (and the one that's currently right above it needs to move to the function below it).

robzienert commented on actions.go:
```diff
@@ -15,6 +15,52 @@ import (
+			"pipeline": pipelineName,
+			"monitor": 	monitor,
+			"retries": 	numRetries,
+		}).Info("Executing Pipeline...")
```
Run `go fmt`

robzienert commented on actions.go:
```diff
@@ -15,6 +15,52 @@ import (
+		if err != nil {
+			return errors.Wrapf(err, "couldn't execute pipeline")
+		}
+		fmt.Printf("Ref task id: %s\n", resp.Ref)
```
And switch all `fmt.Printf` to using logrus.

robzienert commented on spinnaker/http.go:
```diff
@@ -100,7 +100,7 @@ func (c *client) postJSON(url string, body interface{}) (resp *http.Response, re
 		return nil, nil, errors.Wrap(err, "marshaling body to json")
 	}
-
+	logrus.WithField("url", url).Debug("http post")
```
Needed?

mikerowehl committed:
Cleanup a few formatting and output errors:

- Run go fmt on some files that got missed
- Cleanup fmt.Printf statements and make them logrus
- Remove some useless logging

mikerowehl commented:
Ugh, sorry Rob. I thought that branch was clean, but apparently not. Fixed up all that stuff and rebased so it should be current. Thanks!

benjaminws commented:
Getting to the point where this could be useful for our workflow. Thoughts on merging? I realize there's a new CLI on the horizon, but this tool provides a nice stop-gap.

robzienert commented:
Forgot to merge. My b.
